src/Compilers/VisualBasic/Portable/Binding/Binder_Query.vb

```diff
@@ -1897,7 +1896,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Dim keysSelector = keysLambdaBinder.BindExpressionRangeVariables(keys, True, groupBy,
                                                                              keysRangeVariables, diagnostics)
 
-            Dim keysLambda As New BoundQueryLambda(groupBy,
+            Dim keysLambda As New BoundQueryLambda(keysSelector.Syntax,
```

Comment:
It has been updated with the same reason in C#: make it clear to indicate what syntax are associated with the lambda. In the current implementation, different lambdas refer to the same syntax node. The whole purpose of this change is to assign different syntax nodes to different lambdas. In this case, groupBy refers the whole syntax node (GroupByClauseSyntax). Now, keysSelector.Syntax olny refers to a key node. In addition, the below change around "into" mapping would do the same thing.